src/commands/appSettings/filterDownloadAppSettings.ts

```diff
@@ -4,65 +4,61 @@
 import { ext } from "../../extensionVariables";
 import { localize } from "../../localize";
 
 export async function filterDownloadAppSettings(context: IActionContext, sourceSettings: { [key: string]: string }, destinationSettings: { [key: string]: string }, destinationSettingsToIgnore: string[], destinationName: string): Promise<void> {
-    let suppressPrompt: boolean = false;
-    let overwriteSetting: boolean = false;
-    let overwriteSettingsToIgnore: boolean = false;
 
     const addedKeys: string[] = [];
     const updatedKeys: string[] = [];
     const userIgnoredKeys: string[] = [];
     const matchingKeys: string[] = [];
-    const securitySettingsIgnored: string[] = [];
-    const listOfSettingsToIgnore: string[] = ["AzureWebJobsStorage", "WEBSITE_CONTENTAZUREFILECONNECTIONSTRING", "WEBSITE_CONTENTSHARE"];
-
-    const yesToAll: vscode.MessageItem = { title: localize('yesToAll', 'Yes to all') };
-    const noToAll: vscode.MessageItem = { title: localize('noToAll', 'No to all') };
 
+    const listOfSettingsToIgnore: string[] = ["AzureWebJobsStorage", "WEBSITE_CONTENTAZUREFILECONNECTIONSTRING", "WEBSITE_CONTENTSHARE"];
+    const userChosenSettings: string[] = [];
+    const options: vscode.QuickPickItem[] = [];
     destinationSettingsToIgnore.length = 0;
 
-    for (const key of Object.keys(sourceSettings)) {
-        if (listOfSettingsToIgnore.includes(key)) {
-            if (!suppressPrompt) {
-                const message: string = localize('overwriteSettingsToIgnore', 'Setting "{0}" contains secrets. Download?', key);
-                const result: vscode.MessageItem = await context.ui.showWarningMessage(message, { modal: true }, DialogResponses.yes, yesToAll, DialogResponses.no, noToAll);
-                overwriteSettingsToIgnore = result === DialogResponses.yes || result === yesToAll;
-                suppressPrompt = result === yesToAll || result === noToAll;
-            }
-            if (overwriteSettingsToIgnore) {
-                updatedKeys.push(key);
-                destinationSettings[key] = sourceSettings[key];
-            } else {
-                securitySettingsIgnored.push(key);
-                destinationSettings[key] = "_REDACTED_";
-                destinationSettingsToIgnore.push(key);
-            }
+    for (const element of Object.keys(sourceSettings)) {
+        if (listOfSettingsToIgnore.includes(element)) {
+            options.push({
+                label: element
+            });
         }
         else {
-            if (destinationSettings[key] === undefined) { // Have an explicit check for undefined as empty settings should not pass the condition
+            options.push({
+                label: element,
+                picked: true
+            });
+        }
+    }
+
+    const result = await context.ui.showQuickPick(options, { placeHolder: 'Please select the app settings you would like to download:', canPickMany: true });
+
+    while (result.length != 0) {
```

Comment:
why do a pop and than assign? why not just do something like. You want to try to minimize creating temporary variables. Also try your best to declare variables when needed. 

```
const userChosenSettings: string[] = result.map(item => item.label);
```

Reply:
actually

```
const userChosenSettings: string[] = result ?  result.map(item => item.label) : [];
```

Reply:
I was looking for something like this, thank you!!